test_ownership_released_on_reap is flaky

`tests/unit/test_orchestrator.py::TestFileOwnership::test_ownership_released_on_reap` fails intermittently on an unchanged tree.

Measured on 2026-08-27, same checkout, same command, `-p no:randomly`:

| tree | runs | failures |
|---|---|---|
| a feature branch | 17 | 1 |
| `main` | 8 | 0 |

The failure is always the same assertion:

```
assert "src/owned.py" not in orch._file_ownership
AssertionError: assert "src/owned.py" not in {"src/owned.py": "backend-stale"}
tests/unit/test_orchestrator.py:2092
```

The reap itself happens (the `Reaping stale agent` warning is emitted); the ownership entry is what does not always get released. So the suspect is a race or an ordering dependency inside the reap path rather than the reap decision.

**Where to start.** `_reap_and_cleanup_session` in `src/bernstein/core/tasks/task_lifecycle.py` is what the tick reaches; find the point where `_file_ownership` is cleared and work out what can skip it. The test builds its orchestrator with an `httpx.MockTransport` that 404s anything unmatched, so an unmatched request turning into an early return is one plausible shape.

**What proves the fix.** The test passing 50 consecutive times on an unchanged tree, checked by a loop, not by reading the code. A one-line `for i in $(seq 1 50)` around the single test id is enough.

**Out of scope.** Do not quarantine, skip, or add a retry marker - a flaky reap test is reporting something real about the reap path. Do not touch the other tests in `TestFileOwnership`.

**Decision left to you.** If the cause turns out to be the test setup rather than the product code, say so on the issue before rewriting the test - that would mean the ownership release is fine and the fixture is racing itself, which is a different fix.

Ask here if anything is unclear.

--- a/tests/unit/test_orchestrator.py
+++ b/tests/unit/test_orchestrator.py
@@ -2043,8 +2043,17 @@ class TestFileOwnership:
         r2 = orch.tick()
         assert len(r2.spawned) == 1  # no conflict, spawns fine
 
+    # The reap path calls agent_lifecycle's own local _is_process_alive, not this
+    # one; patch both bindings so the outcome no longer depends on whether the
+    # real PID 99 is alive on the host (same fix as #1476's sibling test).
+    @patch("bernstein.core.agents.agent_lifecycle._is_process_alive", return_value=False)
     @patch("bernstein.core.agent_recycling._is_process_alive", return_value=False)
-    def test_ownership_released_on_reap(self, _mock_alive: MagicMock, tmp_path: Path) -> None:
+    def test_ownership_released_on_reap(
+        self,
+        _mock_alive: MagicMock,
+        _mock_alive_lifecycle: MagicMock,
+        tmp_path: Path,
+    ) -> None:
         """File ownership released when a stale agent is reaped."""
         transport = _mock_transport(
             {

